rclcpp/include/rclcpp/experimental/timers_manager.hpp

```diff
@@ -496,7 +495,14 @@ class TimersManager
     static bool timer_greater(TimerPtr a, TimerPtr b)
     {
       // TODO(alsora): this can cause an error if timers are using different clocks
-      return a->time_until_trigger() > b->time_until_trigger();
```

Comment:
I see, this seems a bug I introduced when porting the code to rclcpp.
in our internal version, we added a new function `next_call_time` which takes into account cancellations
https://github.com/irobot-ros/rclcpp/blob/galactic-events-executor/rclcpp/src/rclcpp/timer.cpp#L145-L164

Reply:
I think even if `next_call_time` was used, the bug would still manifest -- that would cause a situation where we wait on `timers_cv_.wait(lock, [this]() {return timers_updated_;});`, which won't return until a timer is manually reset.

An alternative fix that we considered was modifiying `rcl` and the timer object to support a cancellation callback, which would then be used similarly to how `set_on_reset_callback` is used in `TimersManager`. If we could trigger a callback upon the timer being reset, then I think that would allow us to break the condition and then re-order the heap, likely fixing the issue.

Reply:
Ok, good point.
I think we should go ahead with the current fix for now. Adding a new callback for timers cancelled is doable, but it's a larger effort.
Any thoughts @mauropasse ?

Reply:
We ended up reverting this diff here in the `timer_greater` function because we discovered it was no longer necessary for the bug fix